src/measure_theory/prod_group.lean

```diff
@@ -0,0 +1,190 @@
+  intro h2E, apply measure_null_of_measure_inv_null hμ (measurable_inv hE), convert h2E using 2,
+  exact set.inv_inv
+end
+
+lemma measurable_measure_mul_right {E : set G} (hE : is_measurable E) :
+  measurable (λ x, μ ((λ y, y * x) ⁻¹' E)) :=
+begin
+  suffices :
+    measurable (λ y, μ ((λ x, (x, y)) ⁻¹' ((λ z : G × G, (1, z.1 * z.2)) ⁻¹' set.prod univ E))),
+  { convert this, ext1 x, congr' 1 with y : 1, simp },
+  apply measurable_measure_prod_mk_right,
+  exact measurable_const.prod_mk (measurable_fst.mul measurable_snd) (is_measurable.univ.prod hE)
+end
+
+lemma lintegral_lintegral_mul_inv (hμ : is_mul_left_invariant μ) (hν : is_mul_left_invariant ν)
+  (f : G → G → ennreal) (hf : measurable (uncurry f)) :
+  ∫⁻ x, ∫⁻ y, f (y * x) x⁻¹ ∂ν ∂μ = ∫⁻ x, ∫⁻ y, f x y ∂ν ∂μ :=
+begin
+  have h2f : measurable (uncurry $ λ x y, f (y * x) x⁻¹) :=
+  hf.comp ((measurable_snd.mul measurable_fst).prod_mk measurable_fst.inv),
+  simp_rw [lintegral_lintegral h2f, lintegral_lintegral hf],
+  conv_rhs { rw [← map_prod_mul_inv_eq hμ hν] },
+  symmetry, exact lintegral_map hf ((measurable_snd.mul measurable_fst).prod_mk measurable_fst.inv)
+end
+
+lemma measure_mul_right_null (hμ : is_mul_left_invariant μ) {E : set G} (hE : is_measurable E)
+  (y : G) : μ ((λ x, x * y) ⁻¹' E) = 0 ↔ μ E = 0 :=
+begin
+  rw [← measure_inv_null hμ hE, ← hμ y⁻¹ (measurable_inv hE),
+    ← measure_inv_null hμ (measurable_mul_right y hE)],
+  convert iff.rfl using 3, ext x, simp,
+end
+
+lemma measure_mul_right_ne_zero (hμ : is_mul_left_invariant μ) {E : set G} (hE : is_measurable E)
+  (h2E : μ E ≠ 0) (y : G) : μ ((λ x, x * y) ⁻¹' E) ≠ 0 :=
+(not_iff_not_of_iff (measure_mul_right_null hμ hE y)).mpr h2E
+
+/-- A technical lemma relating two different measures. This is basically [Halmos, §60 Th. A].
+  There seems to be a gap in the last step of the proof in [Halmos].
+  In the last line, the equality `g(x⁻¹)ν(Ex⁻¹) = f(x)` holds if we can prove that
+  `0 < ν(Ex⁻¹) < ∞`. The first inequality follows from §59, Th. D, but I couldn't find the second
+  inequality. For this reason, we use a compact `E` instead of a measurable `E` as in [Halmos], and
+  additionally assume that `ν` is a regular measure (we only need that it is finite on compact
+  sets). -/
+lemma measure_lintegral_div_measure [t2_space G] (hμ : is_mul_left_invariant μ)
+  (hν : is_mul_left_invariant ν) (h2ν : regular ν) {E : set G} (hE : is_compact E) (h2E : ν E ≠ 0)
+  (f : G → ennreal) (hf : measurable f) :
+  μ E * ∫⁻ y, f y⁻¹ / ν ((λ h, h * y⁻¹) ⁻¹' E) ∂ν = ∫⁻ x, f x ∂μ :=
+begin
+  let Em := hE.is_measurable,
+  symmetry,
+  set g := λ y, f y⁻¹ / ν ((λ h, h * y⁻¹) ⁻¹' E),
+  have hg : measurable g := (hf.comp measurable_inv).ennreal_div
+    ((measurable_measure_mul_right Em).comp measurable_inv),
+  simp_rw [← set_lintegral_one, ← lintegral_indicator _ Em,
+    ← lintegral_lintegral_mul (measurable_const.indicator Em) hg],
+  rw [← lintegral_lintegral_mul_inv hμ hν],
```

Comment:
```suggestion
  rw [← set_lintegral_one, ← lintegral_indicator _ Em,
    ← lintegral_lintegral_mul (measurable_const.indicator Em) hg,
    ← lintegral_lintegral_mul_inv hμ hν],
```
It looks cleaner to have just one tactic instead of two doing the same job.

Also, after this one, you have two different goals, so you should use braces (or `swap` and get rid right away of the measurability goal).